### PR TITLE
Remove/optimize some allocations in the REST API ingest handlers

### PR DESCRIPTION
**Pull request by guilload**
### Description
- Remove and optimize some gross allocations.
- Backward compatible renames and type changes.

### How was this PR tested?
`make test-all`

### CODE REVIEW

File: quickwit/quickwit-serve/src/ingest_api/rest_handler.rs [hunk 1]
```diff
@@ -293,45 +274,66 @@ pub fn elastic_bulk_handler(
...
+        doc_batches,
+        commit: commit_type as u32,
     };
     let ingest_response = ingest_service.ingest(ingest_request).await?;
     Ok(ingest_response)
 }
 
+/// Returns a byte slice with leading and trailing ASCII whitespace bytes
+/// removed.
+const fn trim_ascii(line: &[u8]) -> &[u8] {
```
Comment (fmassot) [inline quickwit/quickwit-serve/src/ingest_api/rest_handler.rs:312]: `trim_ascii_whitespace` is more informative

File: quickwit/quickwit-serve/src/ingest_api/rest_handler.rs [hunk 2]
```diff
@@ -498,25 +500,6 @@ mod tests {
         universe.assert_quit().await;
     }
 
-    #[tokio::test]
-    async fn test_ingest_api_bulk_request_returns_400_if_malformed_source() {
```
Comment (fmassot) [inline quickwit/quickwit-serve/src/ingest_api/rest_handler.rs:502]: nitpick: we could keep a test on a malformed action

**Review by fmassot — APPROVED** [review]

File: quickwit/quickwit-serve/src/ingest_api/rest_handler.rs [hunk 3]
```diff
@@ -293,45 +274,66 @@ pub fn elastic_bulk_handler(
...
+        } else {
+            break;
+        }
+    }
+    bytes
+}
+
+fn lines(body: &Bytes) -> impl Iterator<Item = &[u8]> {
+    body.split(|byte| byte == &b'\n')
+        .map(trim_ascii)
```
[the remark this answers is not in the document]
Comment (guilload) [reply quickwit/quickwit-serve/src/ingest_api/rest_handler.rs:333]: No, `serde_json` is perfectly happy with untrimmed "lines". So I'm ok with removing this step.

File: quickwit/quickwit-serve/src/ingest_api/rest_handler.rs (hunk 2 above)
Comment (guilload) [reply quickwit/quickwit-serve/src/ingest_api/rest_handler.rs:502]: Good point.

**Pull request merged**